### PR TITLE
test: Added appstream workspace tests

### PR DESCRIPTION
**Pull request by nguyen102**
Issue #, if available:

Description of changes:
Added appstream workspace tests: linux, windows, sagemaker

For each test we do the following steps:
1. Launch workspace
2. Click through the connect UI to get the URL/IP Address
3. For Sagemaker and Linux, we check that the URL/IP address can't be connected to from the internet
4. For all workspaces we click the `Connect` button to check that a new AppStream browser window would be open. Note: we stub the opening of the windows, so the windows don't actually open. We just log the window open API was called.
Checklist:

- [x] Have you successfully deployed to an AWS account with your changes?
- [x] Have you written new tests for your core changes, as applicable?

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: main/end-to-end-tests/checkConnection.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// This file is used for checking whether an IP Address is reachable on a particular port
+// eslint-disable-next-line import/no-extraneous-dependencies
+const tcpp = require('tcp-ping');
+
+const [ipAddress, port] = process.argv.slice(2);
+(async () => {
+  const tcpPingPromise = (ipAddr, portForIp) =>
+    new Promise((resolve, reject) => tcpp.probe(ipAddr, portForIp, (err, data) => (err ? reject(err) : resolve(data))));
+  const response = await tcpPingPromise(ipAddress, port);
+  console.log(response);
```
Comment (SanketD92) [inline main/end-to-end-tests/checkConnection.js]: This is logging ping response without context. Let's remove the `console.log` statement, or add info to make the content readable.

Comment (SanketD92) [reply main/end-to-end-tests/checkConnection.js]: I see stdout is being used for test verification. Could we instead have this function return a boolean, and use that for `expect` checks?

Comment (nguyen102) [reply main/end-to-end-tests/checkConnection.js]: Updated to check by return code

**Review by SanketD92 — COMMENTED** [review]
Looks good. Needs to resolve conflicts.

**Comment by nguyen102** [comment]
Updated to resolve the lock file conflict

**Review by SanketD92 — APPROVED** [review]

**Pull request merged**